src/main-app.js

```diff
@@ -74,6 +81,14 @@ export class AddDialog extends LitElement {
     } catch (err) {
       this._snackbar.close();
 
+      const permission = await navigator.permissions.query({ name: 'nfc' });
+      if (permission.state == 'denied' && err.name == "NotAllowedError") {
+        this._snackbar.labelText = "NFC permission is denied, please grant it in browser settings!";
+        this._actionBtn.textContent = "";
+        this._snackbar.open();
+        return;
```

Comment:
if we return here don't you miss adding some event listeners? I mean even if I grant in browser settings afterwards, will it work without reload?

Reply:
We can add an event listener to monitor permission change, but if the permission change is from push operation and user doesn't enable the scan operation before, should we enable the scan operation as well? i.e. we enable scan operation by default as long as nfc permission is granted.

Reply:
That sounds fine

Reply:
Thanks, done.